Dockerfile

```diff
@@ -337,6 +330,27 @@ rmdir "$coverdir/helpers"
 exit $ecode
 EOF
 
+# dlv builds delve for debug variant images
+FROM gobuild-base AS dlv
+ARG DELVE_VERSION
+ARG TARGETPLATFORM
+RUN --mount=target=/root/.cache,type=cache\
+    --mount=target=/go/pkg/mod,type=cache <<EOT
+  set -ex
+  mkdir /out
+  if [ "$(xx-info os)" = "freebsd" ]; then
+    echo "WARN: dlv requires cgo enabled on FreeBSD, skipping: https://github.com/moby/buildkit/pull/5497#issuecomment-2462031339"
+    exit 0
+  fi
+  xx-go install "github.com/go-delve/delve/cmd/dlv@${DELVE_VERSION}"
+  if ! xx-info is-cross; then
```

Comment:
Why did you change the `GOBIN` solution?

Reply:
We can't install cross-compiled binaries when GOBIN is set: https://github.com/golang/go/blob/84e58c84fdbe4681b4cbfeb615aaba3437c8ac85/src/cmd/go/internal/work/build.go#L771-L772